app/services/organizations/default_pet_task_service.rb

```diff
@@ -0,0 +1,22 @@
+# Service object that creates the default pet tasks for a newly created Pet
+class Organizations::DefaultPetTaskService
+  def initialize(pet)
+    @pet = pet
+    @default_pet_tasks = DefaultPetTask.all
+  end
+
+  def create_tasks
```

Comment:
You can replace this with a single/bulk query for a faster insert:

default_tasks = @default_pet_tasks.map do |task|
  {
    pet_id: @pet.id
    name: task.name,
    description: task.description,
    completed: false
   }
end

Task.create(default_tasks)

Reply:
Thank you for that. I totally forgot about passing an array of hashes to `create`. 
From what I have read it is not a single query, however, and I don't think is any more performant than what I had. It still treats each create as a separate insert statement. Though it does read a little cleaner so I will use it.

Reply:
There is the `insert_all` method that does bulk insert in a single query: https://edgeapi.rubyonrails.org/classes/ActiveRecord/Persistence/ClassMethods.html#method-i-insert_all-21
No objects are created and validations are skipped. Given the number of default tasks is probably unlikely to exceed, say, 20, I think using `create` will suffice. Though, open to other views on that.